bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -1564,6 +1567,14 @@ void handleBookieFailure(final Map<Integer, BookieSocketAddress> failedBookies)
 
         int curNumEnsembleChanges = numEnsembleChanges.incrementAndGet();
 
+        // when the ensemble changes are too frequent, close handle
+        if (curNumEnsembleChanges > maxNumEnsembleChanges){
+            if (LOG.isDebugEnabled()) {
+                  LOG.debug("touch maxNumEnsembleChanges");
```

Comment:
Can you provide a more meaningful log statement here? E.g. "Ledger {} reaches max num ensemble changes {}".